src/common.ts

```diff
@@ -39,6 +39,7 @@ export const enum Context {
   OptionsSpecDeviation = 1 << 29,
   AllowEscapedKeyword = 1 << 30,
   OptionsUniqueKeyInPattern = 1 << 31,
+  OptionsHTMLCommentModule = 1 << 32,
```

Comment:
If you check the value of `1 << 32` in node, it returns `1`. This is due to JS uses 32 bits integer for `<<` operator (store float 1 in 32 bits integer, do shift, then convert back to float).
So `1 << 32` is same as OptionNext `1 << 0`.
We have exhausted the 32 bits for all options. I am not sure how to add one more option, if we give up binary mask, we will loose performance for option check.
As @fisker suggested, putting it behind webcompat might be an option.

Reply:
What other parsers do?

Reply:
> What other parsers do?

Babel throws when `sourceType: module`,  only parse when `sourceType` set to `script` or `unambiguous`.

Reply:
> We have exhausted the 32 bits for all options. I am not sure how to add one more option, if we give up binary mask, we will lose performance for option check.

I feel like losing performance is an acceptable trade-off for more options. Perhaps the context could be an object, providing similar-if not better performance.

Reply:
If this PR isn't adding bulk, I will attempt to make the context an object. I believe this will have benefits beyond adding more options. Doing a binary expression when checking an option may be less efficient when compared to reading a property.